### PR TITLE
Replace `--console=plain` gradle flag

### PR DESCRIPTION
**Pull request by nicknovitski**
gradleSetupHook now includes `--console plain`: https://github.com/NixOS/nixpkgs/commit/c12b2a0b196a42a20949cf648c86acfbe6418ad3#diff-05edf78db91e194716431e8387e022e3734cf39e3ecfb550373fe9378ffa51ecR7

Gradle errors when the same flag is passed twice: https://github.com/gradle/gradle/issues/30913

Although maybe it should only be dropped if it's already present in `gradleFlagsArray`?

### CODE REVIEW

**Comment by raphiz** [comment]
Thanks for the PR!

Yeah, ideally, it should only be dropped if it's already present in `gradleFlagsArray` but I guess that's tricky to figure out.

However, I think there is a simple workaround that we can use. Gradle allows us to set the relevant [Gradle Property](https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_configuration_properties) not only via this command line switch but also as JVM argument . The `--console` switch takes precedence over it - but setting the JVM argument too acts as a fallback. 

Long Story short: Instead of dropping `--console=plain`, we could add ` -Dorg.gradle.console=console` instead.

What do you think? Could you verify this workaround?

**Comment by nicknovitski** [comment]
Ooh, i will try that.

**Comment by nicknovitski** [comment]
Yep, I was able to make this work on the example applications with both `nixos-24.05` and `nixos-24.11` and only small changes.

**Comment by raphiz** [comment]
Nice, thanks!

**Comment by philipwilk** [comment]
Hi, I've updated my flake to rev 6392bb6 but it still has the same error when I try to build/run. Is there any other updates/modifications I need to make? I've got it [in this flake](https://github.com/philipwilk/templates/tree/main/java) because i want to use it as a template (like ``nix flake init -t git+https://git.fogbox.uk/templates?ref=main#java``) but I can't right now haha.

**Comment by raphiz** [comment]
@philipwilk interesting... :thinking: Thanks for reporting! I have seen that you have made further adjustments since your last comment. Does the problem still exist? If so, I'll open a separate ticket.

**Comment by philipwilk** [comment]
@raphiz  Idk why but it kinda just started working around the same time as when I added ``inherit jdk gradle``, I can't reproduce it now so I wouldn't worry about it.

Also side note I had to modify these two options to get hello world working that weren't shown in the example (I didn't modify any of gradle's dirs afaik, I just applied the "rules" listed in the readme).
```
buildTask = "installDist"; # (default ":installDist")
installLocation = "app/build/install/*/"; # (default "build/install/*/")
```
